meson.build

```diff
@@ -29,6 +29,10 @@ substs.set('PROJECT_VERSION',      meson.project_version())
 # This is to be used instead of meson.source_root(), as the latter will return
 # the wrong result when systemd is being built as a meson subproject
 project_source_root = meson.current_source_dir()
+relative_source_path = run_command('realpath',
+                                   '--relative-to=@0@'.format(meson.current_build_dir()),
+                                   project_source_root).stdout().strip()
```

Comment:
doesn't meson have functionality for this built-in? Sounds like something it should really have

Reply:
Should've would've could've ... but doesnt've ;). 
I mean we can lobby meson to provide it, but if that happens, it'll be a while, so we'll need a local workaround anyway.